x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -246,9 +248,9 @@ export class FleetPlugin
           // since it would run this func on *every* req (other plugins, CSS, etc)
           registerLimitedConcurrencyRoutes(core, config);
           registerAgentRoutes(router, config);
-          registerEnrollmentApiKeyRoutes(router);
```

Comment:
I'm not clear why `registerAgentRoutes` doesn't also use the new router. Can you add a comment in the code about why it's different from the code around it?

Reply:
I will add a comment 👍 in the agent routes we mix API routes to manage agents from Kibana that need to be called as `superuser` and route for the Elastic agent, we should probably split them in two functions